(SERVER-2193) Add thread-dump endpoint to admin API

Sharpie opened the pull request:
This commit adds a `jruby-pool/thread-dump` endpoint to the admin API.
This endpoint can be used to retrieve a backtrace from each JRuby
interpreter registered to the pool when `-Djruby.management.enabled=true`
has been set in JAVA_ARGS.

Sharpie commented:
Requires puppetlabs/jruby-utils#71 for tests to pass.

Magisus commented:
I've released clj-parent 4.1.1 with the jruby-utils update. You can add that bump to this PR.

Magisus commented:
Looks like it also needs a rebase.

Sharpie commented:
Rebased, updated the `clj-parent` dependency, and added documentation.

Tests are still failing, but should be cleared up by puppetlabs/puppet#7704.

Magisus commented:
Thanks for adding those docs!

I ran the new tests locally with an updated submodule, and they passed. Going to merge this.

justinstoller commented:
Should be retested if/when https://github.com/puppetlabs/puppetserver/pull/2170 lands

justinstoller commented:
I don't suppose you could add a commit to update to the latest version of clj-parent too? I don't want to race this change and #2169.

Sharpie commented:
`clj-parent` bumped to 4.2.0
